### ISSUE TITLE
Wish: Add LogTarget that prints to standard output, and make it default in console config

### ISSUE THREAD

**Issue by fsateler** [yiisoft/yii2#8840]
I have been bitten quite a few times when (for example) cache/flush-all doesn't work because of permissions issue (say, the cache files are owned by www-data), and in turn the error messages are redirected to the app log... which can't be written because it is owned by www-data too.

The console commands should default to printing everything on stderr, and that should include warning/error log messages.

**Comment by mikehaertl** [comment]
Our tiny extension here was written for logging from docker containers but should probably also work in your case:

https://github.com/codemix/yii2-streamlog

**Comment by fsateler** [comment]
@mikehaertl Interesting, will look into that extension. Thanks!

**Comment by schmunk42** [comment]
@fsateler `yii2-streamlog` is a highly reliable extension, we're using it since over 2 years in all projects (>50) and never had any problems with it.

**Comment by samdark** [comment]
The extension is good as many other extensions by codemix team but I'm not sure if it is a good idea to make it part of the core and use it by default.

**Comment by samdark** [comment]
Actually, Yii prints out error messages by default in the command line mode.

**Comment by samdark** [comment]
So I'd need a way to reproduce the case when it's not doing so.

**Comment by schmunk42** [comment]
> The extension is good as many other extensions by codemix team but I'm not sure if it is a good idea to make it part of the core and use it by default.

You should have some kind of extensions which are let's say "recommended, but without official support from the core team", **codemix** has several of those, **creocoder**, **cebe** and **2amigos** also. **dektrium**/user is also widely used and of high quality.

This would take some weight of your shoulders, while also giving the community some hints about which extensions to chose or try first.

Just as an example ;) You don't need to create any more extensions https://github.com/schmunk42/yii2-extension-requests/issues/8#issuecomment-39006307

> Actually, Yii prints out error messages by default in the command line mode.

That's a matter of configuration, i.e. we have this for [all application types/common](https://github.com/dmstr/phd5-app/blob/master/src/config/common.php#L122)

```
        'log' => [
            'targets' => [
                [
                    'class' => 'codemix\streamlog\Target',
                    'url' => 'php://stderr',
                    'levels' => ['error', 'warning'],
                    'logVars' => [],
                ],
            ],
        ],
```

(And info & trace for web in DEV mode)[https://github.com/dmstr/phd5-app/blob/master/src/config/web.php#L25]

**Comment by fsateler** [comment]
@schmunk42 Indeed, that is what I have been using since. And I wouldn't mind the template to come configured as you did (like I do for my projects now). Yii already depends on third-party components, so adding a new one wouldn't bother me.

@samdark Warning messages are not printed as errors:

```sh
composer create-project yiisoft/yii2-app-basic
cd yii2-app-basic
# simulate a cache written by the server
mkdir runtime/cache/test
touch runtime/cache/test/test
sudo chown --recursive www-data:www-data runtime/cache/test
# and then a flush as user
php yii cache/flush-all
 The following cache components were processed:

	* cache (yii\caching\FileCache)

# Cool! everything worked, right? nope
find runtime/cache 
 runtime/cache
 runtime/cache/test
 runtime/cache/test/test
# But yii is a good framework, it logs about such things:
grep warning runtime/logs/app.log
2017-01-10 00:26:46 [-][-][-][warning][yii\caching\FileCache::gcRecursive] Unable to remove file '/tmp/yii2-app-basic/runtime/cache/test/test': unlink(/tmp/yii2-app-basic/runtime/cache/test/test): Permission denied
2017-01-10 00:26:46 [-][-][-][warning][yii\caching\FileCache::gcRecursive] Unable to remove directory '/tmp/yii2-app-basic/runtime/cache/test': rmdir(/tmp/yii2-app-basic/runtime/cache/test): Directory not empty
```

However, that last step would not have worked if the log file was also owned by the `www-data` group, as yii would not have been able to write to it.

**Comment by schmunk42** [comment]
That's strange, have a look at [these build warnings](https://travis-ci.org/dmstr/phd5-template#L316), the warning is shown in the console, with the settings I've linked above. And also written to `console.log`

How is the `app.log` file created, with a `FileLogRoute` or because output get's send to `stderr`?

Do you expect output to be in `error.log` or on your console?

*As a workaround: You might try configuring error output to `stdout`.*

**Comment by fsateler** [comment]
@schmunk42 I was talking when yii does not have the streamlog configured (that is, the default configuration). With the streamlog configuration they are printed to stderr.

**Comment by samdark** [comment]
Need to verify that. If there's no errors outputted then it worth having in the core and using by default.

**Comment by SilverFire** [comment]
I've wrote `StdoutTarget` for asset-packagist.org recently. [sources](https://github.com/hiqdev/asset-packagist/blob/master/src/log/StdoutTarget.php)

[Had to set](https://github.com/hiqdev/asset-packagist/blob/master/src/config/hidev.php#L31-L36) `exportInterval` and `flushInterval` to 1 in order to have messages printed immediately.
